cardinal/ecs/storage/redis/schema.go

```diff
@@ -0,0 +1,29 @@
+package redis
+
+import (
+	"context"
+
+	"github.com/rotisserie/eris"
+)
+
+type SchemaStorage interface {
+	GetSchema(componentName string) ([]byte, error)
+	SetSchema(componentName string, schemaData []byte) error
+}
+
+func (r *Storage) GetSchema(componentName string) ([]byte, error) {
+	ctx := context.Background()
+	schemaBytes, err := r.Client.HGet(ctx, r.schemaStorageKey(), componentName).Bytes()
+	err = eris.Wrap(err, "")
+	if err != nil {
+		return nil, err
+	}
+	return schemaBytes, nil
+}
+
+func (r *Storage) SetSchema(componentName string, schemaData []byte) error {
+	ctx := context.Background()
+	return eris.Wrap(r.Client.HSet(ctx, r.schemaStorageKey(), componentName, schemaData).Err(), "")
+}
+
+var _ NonceStorage = &Storage{}
```

Comment:
This should be moved near the top of nonce.go

Reply:
This change wasn't done because scott requested a complete change in the architecture.